app/src/screens/ActiveCall/hooks.ts

```diff
@@ -108,76 +106,86 @@ const useHangup = (
-        false,
-      ],
-    )
-    .otherwise(() => [() => {}, true, false]);
+/**
+ * Return type of the audio device hook.
+ */
+type AudioDevicesHookReturnType = {
+  selectedType?: TwilioAudioDevice.Type;
+  onPressSpeaker?: () => void;
+  onPressBluetooth?: () => void;
 };
-
 /**
  * Hook for audio device selection button handlers and state.
  * @param activeCall - The active call.
  * @param audioDevices - The available audio devices.
  * @param dispatch - The dispatch function for the Redux store.
  * @returns - Handlers and state for the audio device selection button.
  */
-const useSelectAudioOutputDevice = (
+const useAudio = (
   activeCall: ActiveCall | undefined,
   audioDevices: AudioDevicesState,
   dispatch: ReturnType<typeof useTypedDispatch>,
-) => {
-  const [handle, isDisabled, isActive] = React.useMemo(
-    () =>
-      match<
-        [typeof activeCall, typeof audioDevices],
-        [() => void, boolean, boolean]
-      >([activeCall, audioDevices])
-        .with(
-          [{ info: { state: 'connected' } }, { status: 'fulfilled' }],
-          ([_, au]) =>
-            matchOnAudioDevices(au.audioDevices, au.selectedDevice, dispatch),
-        )
-        .otherwise(() => [() => {}, false, false]),
-    [dispatch, audioDevices, activeCall],
+): AudioDevicesHookReturnType => {
+  const selectedType = React.useMemo((): TwilioAudioDevice.Type | undefined => {
+    if (audioDevices.status !== 'fulfilled') {
+      return;
+    }
+    return audioDevices.selectedDevice?.type;
+  }, [audioDevices]);
+
+  const bindDevice = React.useCallback(
+    (type: TwilioAudioDevice.Type): (() => void) | undefined => {
+      if (audioDevices.status !== 'fulfilled') {
+        return;
+      }
+
+      const device = audioDevices.audioDevices.find((d) => d.type === type);
+      if (typeof device === 'undefined') {
+        return;
+      }
+
+      return () =>
+        dispatch(selectAudioDevice({ audioDeviceUuid: device.uuid }));
+    },
+    [audioDevices, dispatch],
   );
 
-  return { handle, isActive, isDisabled };
+  const selectEarpiece = React.useMemo(() => {
+    return bindDevice(TwilioAudioDevice.Type.Earpiece);
+  }, [bindDevice]);
+
+  const selectSpeaker = React.useMemo(() => {
+    return bindDevice(TwilioAudioDevice.Type.Speaker);
+  }, [bindDevice]);
+
+  const selectBluetooth = React.useMemo(() => {
+    return bindDevice(TwilioAudioDevice.Type.Bluetooth);
+  }, [bindDevice]);
+
+  const onPressSpeaker = React.useMemo(() => {
+    return selectedType === TwilioAudioDevice.Type.Speaker
+      ? selectEarpiece
+      : selectSpeaker;
+  }, [selectEarpiece, selectSpeaker, selectedType]);
+
+  const onPressBluetooth = React.useMemo(() => {
+    return selectedType === TwilioAudioDevice.Type.Bluetooth
+      ? selectEarpiece
+      : selectBluetooth;
+  }, [selectEarpiece, selectBluetooth, selectedType]);
+
+  if (
+    typeof activeCall === 'undefined' ||
+    audioDevices.status !== 'fulfilled'
+  ) {
+    return {};
```

Comment:
No, we shouldn't throw errors in hooks, instead the hook-managed state should reflect that. In this case, the handlers will be undefined since we're returning an empty object. And the component will be disabled since the handlers are undefined.